### PR TITLE
New docker compose

### PR DESCRIPTION
**Pull request by dasawanaka**
New docker compose. 

All the most important information with commands on how to run is on the [README](https://github.com/edmcouncil/html-pages/blob/new-docker-compose/README.md)

### CODE REVIEW

**Review by patrycjamia — CHANGES_REQUESTED** [review]
strapi not run 
```
Use 'docker scan' to run Snyk tests against images to find vulnerabilities and learn how to fix them
Creating html-pages_strapi_1  ... done
Creating html-pages_home_1    ... done
Creating html-pages_general_1 ... done
Attaching to html-pages_strapi_1, html-pages_general_1, html-pages_home_1
: not found| strapi_setup.sh: line 3:
strapi_1   | [INFO] Starting init strapi
strapi_1   | [INFO] Install strapi without run.
general_1  |
general_1  | > html-pages-general@1.0.0 dev
general_1  | > nuxt
general_1  |
home_1     | yarn run v1.22.19
home_1     | $ vue-cli-service serve
general_1  | ℹ [HPM] Proxy created: /fibo/ontology/api  -> https://spec.edmcouncil.org
general_1  | ℹ [HPM] Proxy created: /fibo/ontology/master/v0.1.0/api  -> https://spec.edmcouncil.org
home_1     |  INFO  Starting development server...
general_1  | ℹ Listening on: http://172.19.0.3:3000/fibo/
strapi_1   | npm WARN exec The following package was not found and will be installed: create-strapi-app@4.10.5
general_1  | ℹ Preparing project for development
general_1  | ℹ Initial build may take a while
general_1  | ℹ Discovered Components: .nuxt/components/readme.md
general_1  | ✔ Builder initialized
             [7%] setup (compile)
             [10%] building (0/1 modules)
general_1  | ℹ Compiling Client
             [51%] building (200/269 modules)
'trapi_1   | error: unknown option '--no-run
strapi_1   | npm notice
strapi_1   | npm notice New minor version of npm available! 9.5.1 -> 9.6.6
strapi_1   | npm notice Changelog: <https://github.com/npm/cli/releases/tag/v9.6.6>
strapi_1   | npm notice Run `npm install -g npm@9.6.6` to update!
strapi_1   | npm notice
: not found| strapi_setup.sh: line 6:
strapi_1   | [INFO] Copy database.
': No such file or directoryeate directory './strapi-dashboard/.tmp
': No such file or directorye './strapi-dashboard/.tmp/data.db
: not found| strapi_setup.sh: line 10:
strapi_1   | [INFO] Copy structures.
: not found| strapi_setup.sh: line 13:
strapi_1   | [INFO] Run Strapi.
strapi_1   | npm ERR! code ENOENT
strapi_1   | npm ERR! syscall open
/package.jsonnpm ERR! path /opt/strapi-dashboard
strapi_1   | npm ERR! errno -2
/package.json'pm ERR! enoent ENOENT: no such file or directory, open '/opt/strapi-dashboard
strapi_1   | npm ERR! enoent This is related to npm not being able to find a file.
strapi_1   | npm ERR! enoent
strapi_1   |
strapi_1   | npm ERR! A complete log of this run can be found in:
             [51%] building (264/300 modules)
              DONE  Compiled successfully in 11992ms12:26:15 PM
home_1     |

home_1     |   App running at:
home_1     |   - Local:   http://localhost:8080/
home_1     |
home_1     |   It seems you are running Vue CLI inside a container.
home_1     |   Access the dev server via http://localhost:<your container's external mapped port>/
home_1     |
home_1     |   Note that the development build is not optimized.
home_1     |   To create a production build, run yarn build.
home_1     |
             Build finished at 12:26:15 by 0.000s
general_1  | ✔ Server: Compiled successfully in 28.27s
general_1  | ✔ Client: Compiled successfully in 31.03s
general_1  | ℹ Waiting for file changes
general_1  | ℹ Memory usage: 602 MB (RSS: 1.12 GB)
general_1  | ℹ Listening on: http://172.19.0.3:3000/fibo/

```

**Review by rosolowski — CHANGES_REQUESTED** [review]
For me:

- Home works correctly,
- Ontology view works correctly,
- Strapi works correctly,

The only problem is that articles don't load. Ontology home (http://localhost:8081/fibo/) seems to work when entered directly (through browser address bar) or on refresh, but not when going from ontology view to the article through menu.

When trying to access http://localhost:8081/fibo/page/community
the app makes request to `http://strapi:1337/api/pages?populate[0]=sections&populate[1]=sections.items&populate[2]=sections.items.image&filters[slug][$eq]=community`, I think it should be localhost instead of strapi

Additionally, this is optional but the `ontologyLogo` (in Config single type) could be left empty in the example strapi database, when the field is not set, the app will use the default EDMC logo.

**Review by trojczak — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
+Using docker compose, the pages run without additional user intervention. The script sets up and copies all the basic files necessary to run on its own.
+The script prepare and publish on localhost 3 pages: 
+- `home` - templates on `http:localhost:8080`
+- `general` - templates on `http:localhost:8081/fibo`
+- `strapi` - required service for `general` pages with fibo database on `http://localhost:1337/admin`
+
+To start and rebuild all containers use command:
+
+```
+$ docker-compose up --build -d
```
Comment (przemekgradzki) [inline README.md]:
```suggestion
$ docker compose up --build -d
```

File: README.md [hunk 2]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
+$ docker logs --follow <container name>
+
+# to view specific amount of logs
+$ docker logs --tail <amount> <container name>
+```
+
+Killing containers takes one command: 
+
+```
+$ docker-compose down
```
Comment (przemekgradzki) [inline README.md]:
```suggestion
$ docker compose down
```

File: README.md [hunk 3]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
+```
+You could also run docker-compose without detached mode(without -d). If so, you'll just use '^C' to kill all containers.
```
Comment (przemekgradzki) [inline README.md]:
```suggestion
You could also run `docker compose` without detached mode(without -d). If so, you'll just use '^C' to kill all containers.
```

**Review by mereolog — CHANGES_REQUESTED** [review]
In macOS Ventura:
1. `docker container logs --follow html-pages_general_1` shows some problems
```
> html-pages-general@1.0.0 dev
> nuxt

ℹ [HPM] Proxy created: /fibo/ontology/api  -> https://spec.edmcouncil.org
...
ℹ [Development] Redirecting from /apple-touch-icon-precomposed.png to /fibo/apple-touch-icon-precomposed.png (router.base specified)

 ERROR  [Vue warn]: option "el" can only be used during instance creation with the new keyword.


 ERROR  [Vue warn]: option "el" can only be used during instance creation with the new keyword.


 ERROR  connect ECONNREFUSED 172.21.0.2:1337

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)


 ERROR  connect ECONNREFUSED 172.21.0.2:1337

  at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1278:16)

```
2. We need a default login and password for http://localhost:1337/admin/auth/login.
3. There are problems with logs - see my comments to https://github.com/edmcouncil/onto-viewer/pull/354.

**Review by patrycjamia — CHANGES_REQUESTED** [review]
firewall certificate problem

```
*   Trying 127.0.0.1:8080...
* Connected to localhost (127.0.0.1) port 8080 (#0)
> GET /fibo/ontology/api/stats HTTP/1.1
> Host: localhost:8080
> User-Agent: curl/8.0.1
> Accept: */*
>
< HTTP/1.1 403 Forbidden
< Server: nginx/1.24.0
< Date: Fri, 26 May 2023 09:34:01 GMT
< Content-Type: text/html
< Content-Length: 12208
< Connection: keep-alive
< Cache-Control: no-cache
< X-XSS-Protection: 1; mode=block
< X-Content-Type-Options: nosniff
<
<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta http-equiv="X-UA-Compatible" content="IE=8; IE=EDGE">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
        <style type="text/css">
            body {
                height: 100%;
                font-family: Roboto, Helvetica, Arial, sans-serif;
                color: #6a6a6a;
                margin: 0;
                display: flex;
                align-items: center;
                justify-content: center;
            }
            input[type=date], input[type=email], input[type=number], input[type=password], input[type=search], input[type=tel], input[type=text], input[type=time], input[type=url], select, textarea {
                color: #262626;
                vertical-align: baseline;
                margin: .2em;
                border-style: solid;
                border-width: 1px;
                border-color: #a9a9a9;
                background-color: #fff;
                box-sizing: border-box;
                padding: 2px .5em;
                appearance: none;
                border-radius: 0;
            }
            input:focus {
                border-color: #646464;
                box-shadow: 0 0 1px 0 #a2a2a2;
                outline: 0;
            }
            button {
                padding: .5em 1em;
                border: 1px solid;
                border-radius: 3px;
                min-width: 6em;
                font-weight: 400;
                font-size: .8em;
                cursor: pointer;
            }
            button.primary {
                color: #fff;
                background-color: rgb(47, 113, 178);
                border-color: rgb(34, 103, 173);
            }
            .message-container {
                height: 500px;
                width: 600px;
                padding: 0;
                margin: 10px;
            }
            .logo {
                background: url(https://globalurl.fortinet.net:8010/XX/YY/ZZ/CI/MGPGHGPGPFGHDDPFGGHGFHBGCHEGPFBGAHAH) no-repeat left center;
                height: 267px;
                object-fit: contain;
            }
            table {
                background-color: #fff;
                border-spacing: 0;
                margin: 1em;
            }
            table > tbody > tr > td:first-of-type:not([colspan]) {
                white-space: nowrap;
                color: rgba(0,0,0,.5);
            }
            table > tbody > tr > td:first-of-type {
                vertical-align: top;
            }
            table > tbody > tr > td {
                padding: .3em .3em;
            }
            .field {
                display: table-row;
            }
            .field > :first-child {
                display: table-cell;
                width: 20%;
            }
            .field.single > :first-child {
                display: inline;
            }
            .field > :not(:first-child) {
                width: auto;
                max-width: 100%;
                display: inline-flex;
                align-items: baseline;
                virtical-align: top;
                box-sizing: border-box;
                margin: .3em;
            }
            .field > :not(:first-child) > input {
                width: 230px;
            }
            .form-footer {
                display: inline-flex;
                justify-content: flex-start;
            }
            .form-footer > * {
                margin: 1em;
            }
            .text-scrollable {
                overflow: auto;
                height: 150px;
                border: 1px solid rgb(200, 200, 200);
                padding: 5px;
                font-size: 1em;
            }
            .text-centered {
                text-align: center;
            }
            .text-container {
                margin: 1em 1.5em;
            }
            .flex-container {
                display: flex;
            }
            .flex-container.column {
                flex-direction: column;
            }
        </style>
        <title>Invalid Connection</title>
    </head>
    <body><div class="message-container">
    <div class="logo"></div>
    <h1></h1>
    <h3>This Connection is Invalid. SSL certificate expired.</h3>
    <p>A secure connection to spec.edmcouncil.org cannot be established.</p>
    <p>When you try to connect securely, sites will present trusted identification to prove that you are going to the right place. However, this site's identity can't be verified.</p>
    <table><tbody>
        <tr>
            <td>Site</td>
            <td>spec.edmcouncil.org</td>
        </tr>
        <tr>
            <td>Certificate CN</td>
            <td>*.makolab.com</td>
        </tr>
        <tr>
            <td>Certificate Authority</td>
            <td>Certum Domain Validation CA SHA2</td>
        </tr>
        <tr>
            <td>Certificate Validity</td>
            <td>
                <div>Not Before: Sep 19 13:57:35 2019 GMT</div>
                <div>Not After: Sep 18 13:57:35 2021 GMT</div>
            </td>
        </tr>
        <tr>
            <td>Certificate Chain</td>
            <td><pre>-----BEGIN CERTIFICATE-----

-----END CERTIFICATE-----
</pre></td>
        </tr>
    </tbody></table>
</div></body>
</html>
```

**Comment by przemekgradzki** [comment]
> firewall certificate problem
> 
> ```
> *   Trying 127.0.0.1:8080...
> * Connected to localhost (127.0.0.1) port 8080 (#0)
> > GET /fibo/ontology/api/stats HTTP/1.1
> > Host: localhost:8080
> > User-Agent: curl/8.0.1
> > Accept: */*
> >
> < HTTP/1.1 403 Forbidden
> < Server: nginx/1.24.0
> < Date: Fri, 26 May 2023 09:34:01 GMT
> < Content-Type: text/html
> < Content-Length: 12208
> < Connection: keep-alive
> < Cache-Control: no-cache
> < X-XSS-Protection: 1; mode=block
> < X-Content-Type-Options: nosniff
> <
> <!DOCTYPE html>
> <html lang="en">
>     <head>
>         <meta charset="UTF-8">
>         <meta http-equiv="X-UA-Compatible" content="IE=8; IE=EDGE">
>         <meta name="viewport" content="width=device-width, initial-scale=1">
>         <link href="https://fonts.googleapis.com/css?family=Roboto&display=swap" rel="stylesheet">
>         <style type="text/css">
>             body {
>                 height: 100%;
>                 font-family: Roboto, Helvetica, Arial, sans-serif;
>                 color: #6a6a6a;
>                 margin: 0;
>                 display: flex;
>                 align-items: center;
>                 justify-content: center;
>             }
>             input[type=date], input[type=email], input[type=number], input[type=password], input[type=search], input[type=tel], input[type=text], input[type=time], input[type=url], select, textarea {
>                 color: #262626;
>                 vertical-align: baseline;
>                 margin: .2em;
>                 border-style: solid;
>                 border-width: 1px;
>                 border-color: #a9a9a9;
>                 background-color: #fff;
>                 box-sizing: border-box;
>                 padding: 2px .5em;
>                 appearance: none;
>                 border-radius: 0;
>             }
>             input:focus {
>                 border-color: #646464;
>                 box-shadow: 0 0 1px 0 #a2a2a2;
>                 outline: 0;
>             }
>             button {
>                 padding: .5em 1em;
>                 border: 1px solid;
>                 border-radius: 3px;
>                 min-width: 6em;
>                 font-weight: 400;
>                 font-size: .8em;
>                 cursor: pointer;
>             }
>             button.primary {
>                 color: #fff;
>                 background-color: rgb(47, 113, 178);
>                 border-color: rgb(34, 103, 173);
>             }
>             .message-container {
>                 height: 500px;
>                 width: 600px;
>                 padding: 0;
>                 margin: 10px;
>             }
>             .logo {
>                 background: url(https://globalurl.fortinet.net:8010/XX/YY/ZZ/CI/MGPGHGPGPFGHDDPFGGHGFHBGCHEGPFBGAHAH) no-repeat left center;
>                 height: 267px;
>                 object-fit: contain;
>             }
>             table {
>                 background-color: #fff;
>                 border-spacing: 0;
>                 margin: 1em;
>             }
>             table > tbody > tr > td:first-of-type:not([colspan]) {
>                 white-space: nowrap;
>                 color: rgba(0,0,0,.5);
>             }
>             table > tbody > tr > td:first-of-type {
>                 vertical-align: top;
>             }
>             table > tbody > tr > td {
>                 padding: .3em .3em;
>             }
>             .field {
>                 display: table-row;
>             }
>             .field > :first-child {
>                 display: table-cell;
>                 width: 20%;
>             }
>             .field.single > :first-child {
>                 display: inline;
>             }
>             .field > :not(:first-child) {
>                 width: auto;
>                 max-width: 100%;
>                 display: inline-flex;
>                 align-items: baseline;
>                 virtical-align: top;
>                 box-sizing: border-box;
>                 margin: .3em;
>             }
>             .field > :not(:first-child) > input {
>                 width: 230px;
>             }
>             .form-footer {
>                 display: inline-flex;
>                 justify-content: flex-start;
>             }
>             .form-footer > * {
>                 margin: 1em;
>             }
>             .text-scrollable {
>                 overflow: auto;
>                 height: 150px;
>                 border: 1px solid rgb(200, 200, 200);
>                 padding: 5px;
>                 font-size: 1em;
>             }
>             .text-centered {
>                 text-align: center;
>             }
>             .text-container {
>                 margin: 1em 1.5em;
>             }
>             .flex-container {
>                 display: flex;
>             }
>             .flex-container.column {
>                 flex-direction: column;
>             }
>         </style>
>         <title>Invalid Connection</title>
>     </head>
>     <body><div class="message-container">
>     <div class="logo"></div>
>     <h1></h1>
>     <h3>This Connection is Invalid. SSL certificate expired.</h3>
>     <p>A secure connection to spec.edmcouncil.org cannot be established.</p>
>     <p>When you try to connect securely, sites will present trusted identification to prove that you are going to the right place. However, this site's identity can't be verified.</p>
>     <table><tbody>
>         <tr>
>             <td>Site</td>
>             <td>spec.edmcouncil.org</td>
>         </tr>
>         <tr>
>             <td>Certificate CN</td>
>             <td>*.makolab.com</td>
>         </tr>
>         <tr>
>             <td>Certificate Authority</td>
>             <td>Certum Domain Validation CA SHA2</td>
>         </tr>
>         <tr>
>             <td>Certificate Validity</td>
>             <td>
>                 <div>Not Before: Sep 19 13:57:35 2019 GMT</div>
>                 <div>Not After: Sep 18 13:57:35 2021 GMT</div>
>             </td>
>         </tr>
>         <tr>
>             <td>Certificate Chain</td>
>             <td><pre>-----BEGIN CERTIFICATE-----
> 
> -----END CERTIFICATE-----
> </pre></td>
>         </tr>
>     </tbody></table>
> </div></body>
> </html>
> ```

Bug fixed in dbf2da0d2714e8559d3e27f5e123c686b49928e3

File: README.md [hunk 4]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
 
+## Run with command line
+
+To run individual pages from the command line, look at the README files given above.
+
+## Run with docker
+
+Using docker compose, the pages run without additional user intervention. The script sets up and copies all the basic files necessary to run on its own.
+The script prepare and publish on localhost 3 pages: 
+- `home` - templates on `http:localhost:8080`
```
Comment (mereolog) [inline README.md]: http://localhost:8080

File: README.md [hunk 5]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
+- `general` - templates on `http:localhost:8081/fibo`
```
Comment (mereolog) [inline README.md]: http://localhost:8080/fibo/

File: README.md [hunk 6]
```diff
@@ -9,3 +9,48 @@ For details, go to the appropriate folder and read the README file contained the
...
+- `strapi` - required service for `general` pages with fibo database on `http://localhost:1337/admin`
```
Comment (mereolog) [inline README.md]: http://localhost:8080/fibo/strapi/admin

**Review by patrycjamia — DISMISSED** [review]

**Review by przemekgradzki — APPROVED** [review]

**Review by mereolog — APPROVED** [review]

**Review by patrycjamia — APPROVED** [review]

**Review by rosolowski — APPROVED** [review]

**Pull request merged**